Modules/_functoolsmodule.c

```diff
@@ -281,29 +281,31 @@ partial_new(PyTypeObject *type, PyObject *args, PyObject *kw)
     }
 
     if (pto_kw == NULL || PyDict_GET_SIZE(pto_kw) == 0) {
-        if (kw == NULL) {
-            pto->kw = PyDict_New();
-        }
-        else if (Py_REFCNT(kw) == 1) {
-            pto->kw = Py_NewRef(kw);
-        }
-        else {
-            pto->kw = PyDict_Copy(kw);
-        }
+        pto->kw = PyDict_New();
     }
     else {
         pto->kw = PyDict_Copy(pto_kw);
-        if (kw != NULL && pto->kw != NULL) {
-            if (PyDict_Merge(pto->kw, kw, 1) != 0) {
-                Py_DECREF(pto);
-                return NULL;
-            }
-        }
     }
     if (pto->kw == NULL) {
         Py_DECREF(pto);
         return NULL;
     }
+    if (kw != NULL) {
+        PyObject *key, *val;
+        Py_ssize_t pos = 0;
+        while (PyDict_Next(kw, &pos, &key, &val)) {
+            if (val == phold) {
+                Py_DECREF(pto);
+                PyErr_SetString(PyExc_TypeError,
+                                "keyword Placeholders are not allowed");
```

Comment:
It is `Placeholder`, not `Placeholders`. Consider rewording. For example:

```suggestion
                                "Placeholder cannot be passed as a keyword argument");
```